src/ets/scala/io/iohk/ethereum/ets/blockchain/ScenarioSetup.scala

```diff
@@ -0,0 +1,143 @@
+package io.iohk.ethereum.ets.blockchain
+
+import io.iohk.ethereum.blockchain.sync.EphemBlockchainTestSetup
+import io.iohk.ethereum.domain.Block.BlockDec
+import io.iohk.ethereum.domain.{Account, Address, Block, UInt256}
+import io.iohk.ethereum.ets.common.AccountState
+import io.iohk.ethereum.ledger.{BlockExecutionError, InMemoryWorldStateProxy, LedgerImpl}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import io.iohk.ethereum.nodebuilder.{BlockchainConfigBuilder, ValidatorsBuilder}
+import io.iohk.ethereum.utils.BigIntExtensionMethods._
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.vm.VM
+import org.spongycastle.util.encoders.Hex
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+
+abstract class ScenarioSetup(scenario: BlockchainScenario)
+  extends EphemBlockchainTestSetup
+  with ValidatorsBuilder
+  with BlockchainConfigBuilder {
+
+  val emptyWorld = blockchain.getWorldStateProxy(-1, UInt256.Zero, None)
+
+  override lazy val blockchainConfig = buildBlockchainConfig(scenario.network)
+
+  val ledger = new LedgerImpl(VM, blockchain, blockchainConfig)
+
+  def loadGenesis(): Block = {
+    val genesisBlock = scenario.genesisRLP match {
+      case Some(rlp) =>
+        val block = rlp.toArray.toBlock
+        assert(block.header == scenario.genesisBlockHeader.toBlockHeader,
+          "decoded genesis block header did not match the expectation")
+        block
+
+      case None =>
+        Block(scenario.genesisBlockHeader.toBlockHeader, BlockBody(Nil, Nil))
+    }
+
+    blockchain.save(genesisBlock)
+    blockchain.save(genesisBlock.header.hash, Nil)
+    blockchain.save(genesisBlock.header.hash, genesisBlock.header.difficulty)
+    genesisBlock
+  }
+
+  def getInitialWorld(): InMemoryWorldStateProxy = {
+    InMemoryWorldStateProxy.persistState(getWorldState(scenario.pre))
+  }
+
+  def getFinalWorld(): InMemoryWorldStateProxy = {
+    InMemoryWorldStateProxy.persistState(getWorldState(scenario.postState))
+  }
+
+  def getBestBlock(): Option[Block] = {
+    val bestBlockNumber = storagesInstance.storages.appStateStorage.getBestBlockNumber()
+    blockchain.getBlockByNumber(bestBlockNumber)
+  }
+
+  private def buildBlockchainConfig(network: String): BlockchainConfig =  network match {
+    case "EIP150"     => new Eip150Config
+    case "Frontier"   => new FrontierConfig
+    case "Homestead"  => new HomesteadConfig
+    case "FrontierToHomesteadAt5" => new FrontierToHomesteadAt5
+    case "HomesteadToEIP150At5" => new HomesteadToEIP150At5
+
+    // Some default config, test will fail or be canceled
+    case _ => new FrontierConfig
+  }
+
+  // Block rlp hex decoding, should be done during test and not parsing json as it can fail
```

Comment:
So should we do something about that?

Reply:
Hex decoding is already moved to test case. I will remove comment as it seems it is misleading.